apps/video/js/view.js

```diff
@@ -206,6 +208,25 @@
     dom.timeSlider.addEventListener('keypress', handleSliderKeypress);
 
     ForwardRewindController.init(dom.player, dom.seekForward, dom.seekBackward);
+
+    // Workaround to fix issue reported in Bug 1214157
+    // We would like to pause video playback when interrupted by competing audio
+    // Audio Channel API throws interruptbegin event after play (Bug 1192748)
+    // If performance.now() - lastPlayTime is small we ignore interruptbegin
+    // and pause video playback only when interrupted by competing audio.
+    // This workaround should be removed with the fix of Bug 1206581
+    dom.player.addEventListener('play', function() {
+      lastPlayTime = window.performance.now();
+    });
+
+    dom.player.addEventListener('mozinterruptbegin', function() {
+      // Ignore mozinterruptbegin event if it immediately follows play event
+      if ((window.performance.now() - lastPlayTime) > 500) {
```

Comment:
While testing updated patch, for view activity I noticed that when we first call showPlayer - andl play() inside it, we are receiving events in sequence
https://github.com/mozilla-b2g/gaia/blob/master/apps/video/js/view.js#L342
1) play
2) interrupt begin
3) inetrrupt end
4) interrupt begin
5) interrupt end
I have checked and this happen only when play is called from inside showPlayer and not for subsequent dom.player.play call.
The second inetrruptbegin call is getting called in time frame > 50 ms and was pausing video when we first open a video in view activity.  
I propose two solutions
a) Increased the time to 500 ms to ignore this second interruptbegin call and fix this bug for view activity
b) Since this bug in view activity is an edge case and we don't update view activity with this work around.